toolchain/check/decl_state.h

```diff
@@ -0,0 +1,80 @@
+// Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+// Exceptions. See /LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#ifndef CARBON_TOOLCHAIN_CHECK_DECL_STATE_H_
+#define CARBON_TOOLCHAIN_CHECK_DECL_STATE_H_
+
+namespace Carbon::Check {
+
+class KeywordModifierSet {
+ public:
+  enum RawEnum {
+    // At most one of these access modifiers allowed for a given declaration,
+    // and if present it must be first:
+    Private = 1 << 0,
+    Protected = 1 << 1,
+
+    // At most one of these declaration modifiers allowed for a given
+    // declaration:
+    Abstract = 1 << 2,
+    Base = 1 << 3,
+    Default = 1 << 4,
+    Final = 1 << 5,
+    Impl = 1 << 6,
+    Virtual = 1 << 7,
+
+    // Sets of modifiers:
+    Access = Private | Protected,
+  };
+
+  KeywordModifierSet() : keywords(static_cast<RawEnum>(0)) {}
+  KeywordModifierSet(RawEnum k) : keywords(k) {}
```

Comment:
```suggestion
  explicit KeywordModifierSet() : keywords(static_cast<RawEnum>(0)) {}
  explicit KeywordModifierSet(RawEnum k) : keywords(k) {}
```

Note this is particularly relevant for the second constructor, which is otherwise an implicit construction (which if you intend it, should be commented and explicitly marked to suppress the relevant clang-tidy warning).

Reply:
Code has been replaced.